docs/flow: add monitoring docs (#2140)
grafana/agent: docs/flow/monitoring/debugging.md

---

# Debugging

There are two primary ways to debug issues with Grafana Agent Flow:

1. Through the Grafana Agent Flow UI

> 2. Through examining logs

Reviewer: Do we want to also add a third bullet-point, where users might want to check the component-level metrics?

Author: Good idea, I'll write a section for that 👍

Author: Actually, thinking about it, I don't know if component-level metrics helps _debug_ issues (rather, they're more useful for _identifying_ issues). WDYT?

Author: For now, I'm going to leave out checking metrics since I'm not positive that's ever helpful for resolving/debugging issues

## Grafana Agent Flow UI

Grafana Agent Flow includes an embedded UI viewable from Grafana Agent's HTTP
server, which defaults to listening at `http://localhost:12345`.

> The documentation for the [`agent run`][agent run] command describes how to
> modify the address Grafana Agent listens on for HTTP traffic.

This UI provides two views:

* The home page shows a table of running components along with their health.
  Clicking the "View" button navigates to the detail page for that component.
* The "Graph" page shows a graph view of all running components along with
  their health. Clicking a component in the graph navigates to the detail page
  for that component.

> The component detail page shows the following information for each component:

Reviewer 2: Do we need to note that any secret values will be obfuscated in the UI?

Author: Good idea, I'll throw that in a blockquote somewhere as an aside 👍

---

# Debugging

There are two primary ways to debug issues with Grafana Agent Flow:

> 1. Through the Grafana Agent Flow UI

Reviewer 3: ```suggestion
* Through the Grafana Agent Flow UI
```

Reviewer 3: Suggest reserving numbered lists for steps that need a specific order, otherwise use bullets.

Author: Right, the intention here is that people check the UI first before trying to investigate logs.

Author: Should that be more clear in the preceding sentence?

> 2. Through examining logs

Reviewer 3: ```suggestion
* Through examining logs
```

> ## Grafana Agent Flow UI

Author: It makes sense for us to put screenshots somewhere in this section, but I'd like to hold off on doing that until the UI is merged in main just in case details change during review.

Grafana Agent Flow includes an embedded UI viewable from Grafana Agent's HTTP
server, which defaults to listening at `http://localhost:12345`.

> The documentation for the [`agent run`][agent run] command describes how to
> modify the address Grafana Agent listens on for HTTP traffic.

This UI provides two views:

> 1. The home page shows a table of running components along with their health.

Reviewer 3: ```suggestion
* The home page shows a table of running components along with their health.
```

>    Clicking on the "View" button will navigate to the detail page for that

Reviewer 3: ```suggestion
   Clicking the "View" button navigates to the detail page for that
```

   component.

> 2. The "Graph" page shows a graph view of all running components along with

Reviewer 3: ```suggestion
* The "Graph" page shows a graph view of all running components along with
```

>    their health. Clicking on a component in the graph will navigate to the

Reviewer 3: ```suggestion
   their health. Clicking a component in the graph navigates to the
```

   detail page for that component.

> The component detail page will show the following information for each

Reviewer 3: ```suggestion
The component detail page shows the following information for each
```

component:

> 1. The health of the component with a message explaining the health.

Reviewer 3: ```suggestion
* The health of the component with a message explaining the health.
```

> 2. The current evaluated arguments for the component.

Reviewer 3: ```suggestion
* The current evaluated arguments for the component.
```

> 3. The current exports for the component.

Reviewer 3: ```suggestion
* The current exports for the component.
```

> 4. The current debug info for the component (if the component has debug info).

Reviewer 3: ```suggestion
* The current debug info for the component (if the component has debug info).
```

To debug using the UI:

* Ensure that no component is reported as unhealthy.
* Ensure that the arguments and exports for misbehaving components appear
  correct.

[agent run]: {{< relref "../reference/cli/run.md" }}

## Examining logs

Logs may also help debug issues with Grafana Agent Flow.

> To reduce on logging noise, many components log hide debugging info behind

Reviewer 3: ```suggestion
To reduce logging noise, many components hide debugging info behind
```

> debug-level log lines. It is recommended to configure the [`logging

Reviewer 3: ```suggestion
debug-level log lines. It is recommended that you configure the [`logging
```

> block`][logging] to emit debug-level log lines when debugging issues with

Reviewer 3: Is this supposed to be emit or omit?  Just making sure.

Author: Yep, emit is right here: we recommend enabling debug logs so users see everything while debugging. 

Should we pick another word here?

Author: I've changed this to show, LMK if you think that sounds ok :)

Grafana Agent Flow.

The location of Grafana Agent's logs is different based on how it is deployed.

> Refer to the [`logging block`][logging] page for how to find logs for your

Reviewer 3: ```suggestion
Refer to the [`logging block`][logging] page to see how to find logs for your
```

---

# Debugging

There are two primary ways to debug issues with Grafana Agent Flow:

1. Through the Grafana Agent Flow UI
2. Through examining logs

## Grafana Agent Flow UI

Grafana Agent Flow includes an embedded UI viewable from Grafana Agent's HTTP
server, which defaults to listening at `http://localhost:12345`.

> The documentation for the [`agent run`][agent run] command describes how to
> modify the address Grafana Agent listens on for HTTP traffic.

This UI provides two views:

* The home page shows a table of running components along with their health.
  Clicking the "View" button navigates to the detail page for that component.
* The "Graph" page shows a graph view of all running components along with
  their health. Clicking a component in the graph navigates to the detail page

>   for that component.

Author: UI elements should be referenced in bold rather than quotation marks: https://developers.google.com/style/ui-elements#formatting

(cc @karengermond for a heads up, I stumbled on this while reading the style guide)
